src/main/java/wooteco/subway/line/service/LineService.java

```diff
@@ -20,68 +28,104 @@
     private static final Logger log = LoggerFactory.getLogger(LineService.class);
 
     private final LineDao lineDao;
+    private final StationService stationService;
+    private final SectionService sectionService;
 
-    public LineService(LineDao lineDao) {
+    public LineService(LineDao lineDao, StationService stationService, SectionService sectionService) {
         this.lineDao = lineDao;
+        this.stationService = stationService;
+        this.sectionService = sectionService;
     }
 
     @Transactional
-    public LineResponse save(LineCreateRequest lineCreateRequest) {
-        validatesNameDuplication(lineCreateRequest);
-        Line line = new Line(lineCreateRequest.getName(), lineCreateRequest.getColor());
+    public LineCreateResponse create(LineCreateRequest request) {
+        Line line = new Line(request.getName(), request.getColor());
         Line newLine = lineDao.save(line);
-        log.info("{} 노선 생성 성공", newLine.getName());
-        return new LineResponse(newLine.getId(), newLine.getName(), newLine.getColor());
+        Section newSection = addInitialSection(request, newLine);
+        return new LineCreateResponse(newLine, newSection);
     }
 
-    private void validatesNameDuplication(LineCreateRequest lineCreateRequest) {
-        lineDao.findByName(lineCreateRequest.getName())
-                .ifPresent(l -> {
-                    throw new LineDuplicatedNameException();
-                });
+    private Section addInitialSection(LineCreateRequest request, Line newLine) {
+        Station upStation = stationService.findById(request.getUpStationId());
+        Station downStation = stationService.findById(request.getDownStationId());
+        if (upStation.equals(downStation)) {
+            throw new SubwayException("같은 역을 등록할 수 없습니다.");
+        }
+        Section section = new Section(newLine, upStation, downStation, request.getDistance());
+        return sectionService.add(section);
     }
 
-    public LineResponse findBy(Long id) {
-        Line newLine = findById(id);
-        log.info("{} 노선 조회 성공", newLine.getName());
-        return new LineResponse(newLine.getId(), newLine.getName(), newLine.getColor());
+    @Transactional
+    public void addSection(Long id, SectionCreateRequest request) {
+        Line line = lineSetting(id);
+        Station upStation = stationService.findById(request.getUpStationId());
+        Station downStation = stationService.findById(request.getDownStationId());
+        Section section = new Section(line, upStation, downStation, request.getDistance());
+        line.addSection(section);
+        sectionService.synchronizeDB(line);
     }
 
-    private Line findById(Long id) {
-        return lineDao.findById(id)
-                .orElseThrow(LineNotFoundException::new);
+    private Line lineSetting(Long id) {
+        Line line = lineDao.findById(id);
+        Sections sections = sectionService.findByLine(line);
+        line.updateSections(sections);
```

Comment:
updateSections 이라고 쓰면 라인 객체 내에서 section과 관련된 변경 작업이 있는 것으로 오해합니다
지금은 jdbc를 써야하는 상황이니 어쩔 수 없이 setter로 표현하는게 낫겠네요

```suggestion
        line.setSections(sections);
```